guides/writing-addons/deprecations.md

```diff
@@ -0,0 +1,78 @@
+        false,
+        {
+            id: 'your-deprecation-id',
+            until: '2.0.0',
+            url: 'link-to-your-deprecations-instructions'
+        }
+    );
+    // code that should run for the deprecated method
+}
+// ...
+```
+
+The most important reason to use Ember's built in [`deprecate`](https://api.emberjs.com/ember/release/functions/@ember%2Fapplication%2Fdeprecations/deprecate) method is that the warning will be silenced in production.
+If a customer opens up the console for an app that uses your addon, they shouldn't see anything!
+Developers can configure whether it shows up in other build environments as well, like testing and local development.
+
+## Testing the deprecation warning
+
+Whenever you add a deprecation, it's important to add a couple test cases that would trigger the warning. You should also write tests for the "new" way to use the addon.
+It's a good idea to make these examples match what is in your `README` or API docs, so that you know the advice you are giving users actually works.
+
+If you want to test if the deprecation warning itself fires at the right times, you can use community addons like [ember-qunit-assert-helpers](https://github.com/workmanw/ember-qunit-assert-helpers#emberdeprecate-assertions).
+
+After you install the addon with `ember install ember-qunit-assert-helpers`, you can use the `assert.expectDeprecation()` anywhere in your tests where you already use `qunit`. For examples and more ways to use `expectDeprecation`, see the documentation for that addon.
+
+## Documenting the deprecation
+
+There are a few different ways to document the deprecation.
+First, and most importantly, make sure your deprecation warning itself is easy to understand, clearly states which addon it comes from, and has brief instructions for resolving the deprecation.
+
+You should also add information to your `README`, a `CHANGELOG`, or make a dedicated `DEPRECATIONS.md` file that you can link to in your deprecation warning.
+For larger projects, you may even want to have a separate section in your documentation that is just for deprecations.
+Keep in mind that you don't want your library to look like it is swimming in deprecations, so avoid notes about deprecation caveats in your new-user-facing materials, like quick start guides.
+
+## Removing deprecated features
+
+The best time to remove a deprecated feature is when you do a major version, such as going from `v1.3.4` to `2.0.0`.
+
+Removing a deprecation is a "breaking change" and developers rely on each segment of the version numbers to know whether a dependency upgrade is safe. The first number, the "major" version, is for breaking changes. The second number, the "minor" version, is for adding new features. Lastly, the "patch" version is for bugfixes and refactors. This system is referred to as [SemVer](https://semver.org/), and it is the standard for packages across the JavaScript ecosystem.
```

Comment:
In JS a breaking change is indicated by increasing the first _non-zero_ number in the version. If your addon is above 1.0.0, that means major version bumps, but if your addon is 0.5.0 then 0.6.0 is how you indicate breaking changes.